### DISCUSSION TITLE
Trying to save change to model from TE3 while model is reprocessing will cause TE3 to hang indefinitely

### DISCUSSION THREAD

**Discussion by Javanite** [TabularEditor/TabularEditor3#1057] [Ideas]
Experienced this issue while trying to save a new measure to the model while my coworker was in the middle of a large reprocessing action and TE3 will pop up the "Please Wait - Saving Changes..." box and will never complete which results with me having to kill TE3 with task manager.

Suggested fix is for TE3 to recognize that there is a lock on the model due to reprocessing and throw an error or offer a "cancel" option to the user in case it's taking too long.

![image](https://user-images.githubusercontent.com/19479604/127521624-1868f2e3-58d2-46ef-96ea-f0430540b65a.png)

![image](https://user-images.githubusercontent.com/19479604/127521569-92fc36d5-f8b0-42de-9995-b3231b839518.png)

![image](https://user-images.githubusercontent.com/19479604/127521709-f89a620c-f3a5-4cb7-b161-ea01638c4c51.png)

**Comment by otykier** [comment]
This one is tricky.

1. There is (to my knowledge) no good way to detect existing (blocking) transactions on all types of Analysis Services instances (that would require access to server-admin level DMV's, which we do not have in for example Power BI Premium XMLA).
2. The TOM [Model.SaveChanges](https://docs.microsoft.com/en-us/dotnet/api/microsoft.analysisservices.tabular.model.savechanges?view=analysisservices-dotnet) method cannot be cancelled.

I'm afraid at least for now, the answer will have to be "please coordinate with your coworkers" or "don't use the same database when developing in parallel". Sorry!

**Comment by Javanite** [comment]
I can appreciate that answer, thanks. It would still be nice for TE3 to gracefully handle the timeout, however. If it tries for too long, couldn't you just throw a generic error?

**Comment by seedjay1** [comment]
Help me understand why the TOM-version of the POST /refreshes api call wouldn't suffice here? (it returns error if a processing is already happening) https://docs.microsoft.com/en-us/azure/analysis-services/analysis-services-async-refresh

**Comment by otykier** [comment]
TOM doesn’t have anything equivalent to the async refreshes API. Also, using that http endpoint is not a viable option in general, as it is only available for Azure AS (not on-premise SSAS, not Power BI XMLA). Also, the transaction could be blocked for reasons other than refresh operations.

**Comment by seedjay1** [comment]
I was afraid that would be the answer :(

Oh well, thanks much!
